mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -1024,11 +1054,45 @@ public IAsyncResult BeginAccept (Socket acceptSocket, int receiveSize, AsyncCall
 				AcceptSocket = acceptSocket,
 			};
 
-			QueueSocketAsyncResult (readQ, sockares.Worker, sockares);
+			QueueIOSelectorJob (readQ, sockares.handle, new IOSelectorJob (IOOperation.Read, s => BeginAcceptReceiveCallback ((SocketAsyncResult) s), sockares));
 
 			return sockares;
 		}
 
+		static void BeginAcceptReceiveCallback (SocketAsyncResult sockares)
+		{
+			Socket acc_socket = null;
+			try {
+				if (sockares.AcceptSocket == null) {
+					acc_socket = sockares.socket.Accept ();
+				} else {
+					acc_socket = sockares.AcceptSocket;
+					sockares.socket.Accept (acc_socket);
+				}
+			} catch (Exception e) {
```

Comment:
Because it is simply going to set `DelayedException` on the `SocketAsyncResult`, which is itself going to be thrown when calling `EndAcceptReceive`.